### PR TITLE
More validation for openmetricsvalidator and scrapevalidator

### PR DESCRIPTION
**Pull request by cw9**

### CODE REVIEW

File: src/cmd/openmetricsvalidator/README.md [hunk 1]
```diff
@@ -30,33 +30,9 @@ We are working on updating the tool to bridge the gap between the python client
...
-FAIL test: bad_unit_6
-FAIL test: bad_unit_7
+FAIL test: bad_timestamp_4
+FAIL test: bad_timestamp_5
+FAIL test: bad_timestamp_7
 FAIL test: duplicate_timestamps
-FAIL test: nan
-FAIL test: no_metadata
-FAILED passed=177, failed=28, total_failures=28
+FAILED passed=201, failed=4, total_failures=4
```
Comment (cw9) [inline src/cmd/openmetricsvalidator/README.md:37]: The remaining failures are caused by OpenMetricsParser failing to parse timestamps in NaN, +Inf, -Inf or in nanoseconds

File: src/validator/validator.go [hunk 2]
```diff
@@ -27,28 +53,77 @@ var (
...
+		level: ErrorLevelMust,
+	}
+
+	errMustNotCounterValueDecrease = errorWithLevel{
+		err:   errors.New("counter total MUST be monotonically non-decreasing over time"),
+		level: ErrorLevelMust,
+	}
+
+	errMustCounterValueBeValid = errorWithLevel{
+		err:   errors.New("A Total is a non-NaN and MUST be monotonically non-decreasing over time, starting from 0"),
```
Comment (robskillington) [inline src/validator/validator.go:67]: super nit: Other errors seem to only have a capital first character if it's a "type" (i.e. MetricPoint, MetricFamilies, Label, etc), maybe this should just be "a Total is a non-NaN ..."?

File: src/validator/validator.go [hunk 3]
```diff
@@ -17,8 +18,33 @@ import (
...
 var (
-	errMustNotCounterValueDecrease = errorWithLevel{
-		err:   errors.New("counter total MUST be monotonically non-decreasing over time"),
+	errExampler = errorWithLevel{
+		err:   errors.New("only histogram/gaugehistogram buckets and counters can have exemplars"),
+		level: ErrorLevelMust,
+	}
+
+	errMustNotMixTimestampPresense = errorWithLevel{
+		err:   errors.New("Mix of timestamp presence within a group"),
```
Comment (robskillington) [inline src/validator/validator.go:27]: super nit: Other errors seem to only have a capital first character if it's a "type" (i.e. MetricPoint, MetricFamilies, Label, etc), maybe this should just be "mix of timestamp present or not present within a group"?

**Review by robskillington — APPROVED** [review]
LGTM

**Pull request merged**